src/main/java/tech/allegro/wildsnake/api/ApiRestController.java

```diff
@@ -0,0 +1,51 @@
+package tech.allegro.wildsnake.api;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+import tech.allegro.wildsnake.product.model.Product;
+import tech.allegro.wildsnake.product.repository.ProductRepository;
+
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Created by Szymon on 06.12.2015.
+ */
+
+@RestController
+@RequestMapping("/api/v1")
+public class ApiRestController {
+
+    private ProductRepository productRepository;
+
+    @Autowired
+    public ApiRestController(ProductRepository productRepository) {
+        this.productRepository = productRepository;
+    }
+
+    @RequestMapping(value = "/products", method = RequestMethod.GET)
+    public List<Product> getProducts(
+            @RequestParam(value = "order", required = false) Optional<String> order,
+            @RequestParam(value = "limit", required = false) Optional<Long> limit
+    ) {
+        if (order.isPresent() && limit.isPresent()) {
+            if ("asc".equals(order.get())) {
+                return productRepository.findFirst3ByOrderByIdAsc();
```

Comment:
Why do you think that every end user of your application will always want receive exactly 3 products? :) Test case is only one use case.